rusty_socket_server/src/socket_server.rs

```diff
@@ -29,27 +36,39 @@ impl SocketServer {
                 }
                 Err(e) => {
                     println!("Error: {}", e);
-                    panic!("Sutting Down Server Due to Error in Stream");
+                    panic!("Shutting Down Server Due to Error in Stream");
                 }
             }
         }
     }
 
-    fn handle_connection(&self, mut stream: TcpStream) {
+    pub fn handle_connection(&self, mut stream: TcpStream) {
         let mut buffer = [0; 512];
         let _ = stream.read(&mut buffer);
         let client_request = String::from_utf8_lossy(&buffer);
 
-        let handshake = HandShake::build(&client_request);
+        let handshake = HandShake::perform(&client_request);
 
         match handshake.request {
-            Some(_request) => {
+            Some(_) => {
                 stream
                     .write_all(handshake.response.to_string().as_bytes())
                     .expect("Failed to write success response");
                 stream.flush().expect("Failed to flush success stream");
 
-                self.handle_frames(stream);
+                match self.active_connections.try_lock() {
```

Comment:
Again, I think protocol should be simple implementation. Complexities like this should be handled by implementing application.

Reply:
Yes, I am bit confused cause we did not decide on how the application would implement, this is something we can talk about,